### PR TITLE
Write semantic tokens config with one rule per line using inline tables

### PR DESCRIPTION
**Pull request by krobelus**
In our config, each entry takes up 5 lines.  Since some language
servers (like rust-analyzer) provide a large number of tokens (>30),
this means that a semantic token configuration quickly exceeds the
size of a terminal window, which makes it a bit less convenient to
skim/read/edit.

We can already use [TOML Inline
Tables](https://github.com/toml-lang/toml/blob/master/toml.md#user-content-inline-table)
to use only one line for each token-face-rule.
However, that would mean we'd need to write something like

	semantic_tokens = [ ... ]

which we can only write at the beginning of kak-lsp.toml (because it
must be inside the nameless global TOML table).  I'd like to keep the
semantic tokens configuration at the bottom of this file because unlike
server configuration, this feature is not required for basic operation.

Use "semantic_tokens.rules" instead of "semantic_tokens" so we can
write it as inline table anywhere in the TOML file. The old syntax
continues to be valid.

Potential concerns:
- the "rules" name feels arbitrary, I couldn't come up with a better name.

Closes #558

### CODE REVIEW

**Review by sclu1034 — APPROVED** [review]
> the "rules" name feels arbitrary, I couldn't come up with a better name.

I can't say "Yes, that's the one" here either, but it's not a bad option.
Other candidates in no particular order:
- `config`
- `map`
- `mappings`
- `tokens`
- `faces`
- `list`
- `values`
- `match`
- `matches`

**Comment by krobelus** [comment]
I like `faces` because it sets the output (face) apart from the inputs (token/modifiers).
Might also want to switch the order to face/token/modifiers or token/modifiers/face instead of token/face/modifiers.

**Comment by sclu1034** [comment]
> Might also want to switch the order to face/token/modifiers or token/modifiers/face instead of token/face/modifiers.

I'd prefer the first one, then. Since `modifiers` is optional, it should come last to keep positioning somewhat consistent.

**Review by sclu1034 — APPROVED** [review]

**Comment by krobelus** [comment]
Thanks for the review. I'll tag a release soon; this is the changelog:

Breaking changes:
- [`python-lsp-server`](https://github.com/python-lsp/python-lsp-server) replaces [python-language-server](https://github.com/palantir/python-language-server) as default language server for Python.
- `rust-analyzer` replaces `rls` as default language server for Rust (#578).

Additions:
- New command `lsp-code-action-sync` is the synchronous variant of `lsp-code-action`, suitable for use in `BufWritePre` hooks (#582).
- The code action menu now provides fuzzy (subsequence) filtering but no longer provides `<c-n>`/`<c-p>` due to implementation reasons (b4ee2a3).
- When accepting a completion, any `additionalTextEdits` are now applied. For example rust-analyzer adds import statements this way (2637b26).
- `lsp-handle-progress` gained a default implementation that shows an hourglass (⌛) in the modeline when a language server is busy (f8a8cf8).
- The default configuration for `semantic_tokens` now uses a more condensed syntax (#587).
- Some systems' C libraries ship old Unicode databases, but terminals often bundle newer versions. To avoid graphical glitches, emoji are only used if they are assigned display width 2 by the C library's `wcwidth(3)` (73c2d1a, c3aeb0d).
- The [clangd protocol extension for offset encoding negotiation](https://clangd.llvm.org/extensions.html#utf-8-offsets) is now supported, which means kak-lsp now uses UTF-8 offsets if the server supports them (#485).

Bug fixes:
- `lsp-formatting-sync` now works again with `texlab` instead of blocking forever (regressed in 11.0.0) (a12831a).
- Certain text edits (such as some code actions from `rust-analyzer`) are now applied correctly (35ba8de).

**Comment by sclu1034** [comment]
> Certain text edits [...] are now applied correctly

This sounds like only that subset works, while all others are still broken 😅 
Maybe reword to "Fix certain text edits [...] not being applied correctly"?

**Comment by krobelus** [comment]
On Tue, Jan 25, 2022 at 01:15:34AM -0800, Lucas Schwiderski wrote:
> > Certain text edits [...] are now applied correctly
> 
> This sounds like only that subset works, while all others are still broken 😅 
> Maybe reword to "Fix certain text edits [...] not being applied correctly"?

Yup, thanks. (I hope all text edits work now)
